### PR TITLE
Check docker-compose version for normalizing container names correctly

### LINKED ISSUE #16: rid has broken since docker-compose 1.21.x
docker-compose allows `-` and `_` in project_name https://github.com/docker/compose/pull/5788 since v1.21.0.
But rid does not allow them currently.
https://github.com/creasty/rid/blob/53be326b5f2c6eba07668f18c42f9af79a385aa7/docker/util.go#L8-L18

### PR DESCRIPTION
**Pull request by izumin5210**
To fix #16 

It works fine with docker-compose v1.20.1 and v1.21.2 in my local :v:

### CODE REVIEW

**Comment by izumin5210** [comment]
@creasty plz review 🙏

**Comment by creasty** [comment]
Oh...

File: docker/util.go [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+		// https://github.com/docker/compose/blob/f55c9d42013e8fbb5285bc402d8248a846485217/compose/cli/command.py#L105
+		projectNormalizePattern = regexp.MustCompile(`[^a-z0-9]`)
+	}
+}
+
 var (
-	// https://github.com/docker/compose/blob/f55c9d42013e8fbb5285bc402d8248a846485217/compose/cli/command.py#L105
-	projectNormalizePattern = regexp.MustCompile(`[^a-z0-9]`)
+	// https://github.com/docker/compose/blob/1.21.0/compose/cli/command.py#L132
+	projectNormalizePattern = regexp.MustCompile(`[^-_a-z0-9]`)
```
Comment (creasty) [inline docker/util.go]: Wouldn't it better to define an uninitialized variable here and assign a proper value in one place?

```go
var (
	projectNormalizePattern regexp.Regexp
)

func init() {
	if doesAllowDashAndUnderscore() {
		projectNormalizePattern = ...
	} else {
		projectNormalizePattern = ...
	}
}
```

Comment (izumin5210) [reply docker/util.go]: fixed 🙆

**Review by creasty — APPROVED** [review]
Thanks!

**Pull request merged**